### PR TITLE
[DPC-4402] Add Expired Invitations list to Organization view for AO

### PR DESCRIPTION
**Pull request by chris-ronning-ny**
## 🎫 Ticket

https://jira.cms.gov/browse/DPC-4402

## 🛠 Changes

- Added a table to the list view component for expired invitations
- Query for expired invitations during OrganizationsController#show action
- Filter out expired invitations from the pending invitations query in OrganizationsController#show action

## ℹ️ Context

We want to render a list of expired invitations, so that the AO can keep track and initiate a new invitation if need be.

## 🧪 Validation

- [ ] Unit testing
- [ ] Manual verification -- see screenshot below

Screenshot 2024-11-21 at 12 10 46 PM

### CODE REVIEW

File: dpc-portal/app/components/page/credential_delegate/list_component.html.erb [hunk 1]
```diff
@@ -27,5 +27,18 @@
         <p>There are no pending credential delegates.</p>
       <% end %>
     </div>
+    <div>
+      <h2>Expired invitations</h2>
+      <p>These invites expired. You can resend the invite to give them more time to accept.</p>
```
Comment (chris-ronning-ny) [inline dpc-portal/app/components/page/credential_delegate/list_component.html.erb:32]: This is what is listed in the ticket A/C as "Table description." While this does not exist for the other two tables yet, I verified with design that this is in the correct position. An equivalent description will be added to the other tables in future tickets.

File: dpc-portal/spec/components/page/credential_delegate/list_component_spec.rb [hunk 2]
```diff
@@ -12,14 +12,15 @@
...
-    let(:component) { described_class.new(org, invitations, credential_delegates) }
+    let(:component) { described_class.new(org, invitations, expired_invitations, credential_delegates) }
 
     before do
       render_inline(component)
     end
 
     context 'No credential delegates' do
       let(:invitations) { [] }
+      let(:expired_invitations) { [] }
```
[the remark this answers is not in the document]
Comment (chris-ronning-ny) [reply dpc-portal/spec/components/page/credential_delegate/list_component_spec.rb:23]: I'm doing this for `requests/organization_spec` too.

File: dpc-portal/app/controllers/organizations_controller.rb [hunk 3]
```diff
@@ -19,12 +19,17 @@ def index
   def show
     show_cds = current_user.ao?(@organization)
     if show_cds
+      # Invitation expiration is determined in relation to the `created_at` field; the `status` field will
+      # never be `'expired'`. Therefore, we need to further filter out expired invitations from this query.
       @invitations = Invitation.where(provider_organization: @organization,
                                       invited_by: current_user,
-                                      status: :pending)
+                                      status: :pending).select { |inv| inv.expired? == false }
```
[the remark this answers is not in the document]
Comment (chris-ronning-ny) [reply dpc-portal/app/controllers/organizations_controller.rb:26]: I made this change, and rubocop transformed it into `.reject(&:expired?)`

**Review by jdettmannnava — APPROVED** [review]
LGTM

**Pull request merged**